### PR TITLE
Contacts and account cards

### PR DESCRIPTION
**Pull request by tjukes**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,43 +1,52 @@
...
+
+### Use Case: Gift Cards
 
-- [Lightrail Stripe Gem](https://github.com/Giftbit/lightrail-stripe-ruby)
-- [Lightrail Java Client](https://github.com/Giftbit/lightrail-client-java)
-- [Lightrail-Stripe Java Integration](https://github.com/Giftbit/lightrail-stripe-java)
+**Looking for Lightrail's Drop-In Gift Card Solution?** 
 
-### Gift Cards
+Lightrail's Drop-in Gift Card solution makes it very easy to add gift cards and account credits to your application. The solution is component based, using snippets that can be included for gift card sending and redemption. It requires minimal updates to your existing checkout process. Check out the [Drop-in Gift Card documentation](https://github.com/Giftbit/Lightrail-API-Docs/blob/master/docs/quickstart/drop-in-gift-cards.md#drop-in-gift-cards) to get started.
```
Comment (pushplay) [inline README.md]: I like the motivation for this and what I wonder when looking at this is how it's going to age.  Do we copy and paste this paragraph into the other libraries' readmes?  How many places do we have to change if the Drop In landing page changes?  What about when we have platform-specific integrations?

I think we have a page that lists all our libraries.  What if we copied this text there and then linked to that with something more similar?  eg:

> Not sure if this is the right library for you?  Visit [some link here] for an overview of our integrations.

File: lib/lightrail_client/account.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
+module Lightrail
+  class Account < Lightrail::LightrailObject
+    def self.create(account_params)
```
Comment (pushplay) [inline lib/lightrail_client/account.rb]: I could be misreading this, but it looks like this is establishing an `Account` domain object that can be retrieved or modified.  That's inconsistent with the [JS lib](https://github.com/Giftbit/lightrail-client-javascript/blob/master/src/accounts.ts#L13) where accounts functions return cards or transactions.

The problem here is that the REST API doesn't really have a clear idea what an account is, so it's easy for each lib to do it differently.  I think we should talk about this because I do want to be consistent.

File: lib/lightrail_client/account.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      validated_params = Lightrail::Validator.set_params_for_account_create!(account_params)
+      response = Lightrail::Connection.send :make_post_request_and_parse_response, "cards", validated_params
+      response['card']
+    end
+
+    def self.retrieve_by_shopper_id_and_currency(shopper_id, currency)
```
Comment (pushplay) [inline lib/lightrail_client/account.rb]: Is that the pattern we should be following?  [JS](https://github.com/Giftbit/lightrail-client-javascript/blob/master/src/contacts.ts#L27) uses an object that specifies one of the IDs.

File: lib/lightrail_client/card.rb [hunk 4]
```diff
@@ -26,7 +32,7 @@ def self.get_maximum_value(card_id)
     end
 
     def self.get_details(card_id)
-      response = Lightrail::Connection.make_get_request_and_parse_response("cards/#{card_id}/details")
+      response = Lightrail::Connection.make_get_request_and_parse_response("cards/#{CGI::escape(card_id)}/details")
```
Comment (pushplay) [inline lib/lightrail_client/card.rb]: :+1:

File: lib/lightrail_client/account.rb (hunk 3 above)
Comment (pushplay) [reply lib/lightrail_client/account.rb]: Wait is this saying the Account is the Card or the Contact?  Because if it's either the Account is the Card.  The Account holds the balance.

Comment (tjukes) [reply lib/lightrail_client/account.rb]: The Account is the Card.

File: lib/lightrail_client/account.rb (hunk 2 above)
Comment (tjukes) [reply lib/lightrail_client/account.rb]: This does put 'Accounts' up at the top level, but none of the methods return instances of Accounts. They all return plain hashes, same as other classes & methods.

File: README.md (hunk 1 above)
Comment (tjukes) [reply README.md]: That link is already included, but I can shorten this to make it less likely to need changes.
